src/bytes_mut.rs

```diff
@@ -1732,6 +1740,28 @@ unsafe fn shared_v_to_vec(data: &AtomicPtr<()>, ptr: *const u8, len: usize) -> V
     }
 }
 
+unsafe fn shared_v_to_mut(data: &AtomicPtr<()>, ptr: *const u8, len: usize) -> BytesMut {
+    let shared: *mut Shared = data.load(Ordering::Relaxed).cast();
+
+    if (*shared).is_unique() {
+        let shared = &mut *shared;
+
+        let ptr = vptr(ptr as *mut u8);
+        let cap = len; // It will try to reclaim the buffer on first insert
```

Comment:
This could be improved I think by doing:
1. fetching the original cap
2. calculate the offset of the data from ptr
3. doing cap - offset
I just was not sure it was worth it, let me know

Reply:
I think it makes sense to use a larger cap when possible.